src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithSimpleName.cs

```diff
@@ -63,7 +63,8 @@ public partial struct SyntaxValueProvider
         // changed. CreateSyntaxProvider will have to rerun all incremental nodes since it passes along the
         // SemanticModel, and that model is updated whenever any tree changes (since it is tied to the compilation).
         var syntaxTreesProvider = _context.CompilationProvider
-            .SelectMany((compilation, cancellationToken) => GetSourceGeneratorInfo(syntaxHelper, compilation, cancellationToken))
+            .SelectMany(static (compilation, _) => compilation.CommonSyntaxTrees)
+            .Select((tree, cancellationToken) => (Tree: tree, Info: tree.GetSourceGeneratorInfo(syntaxHelper, cancellationToken)))
```

Comment:
Both steps here should have a tracking name. And the tests should validate the values in both steps.

Reply:
Good point, will add.